samples/bot-types-of-cards/nodejs/bots/botTypeOfCards.js

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityHandler, MessageFactory, CardFactory } = require('botbuilder');
+const { ActionTypes } = require('botframework-schema');
+const AdaptiveCard = require('../resources/adaptiveCard.json');
+const Office365ConnectorCard = require('../resources/o365ConnectorCard.json');
+const ThumbnailCard = require('../resources/thumbnailCard.json');
+const ListCard = require('../resources/listCard.json');
+const CollectionCard = require('../resources/collectionsCard.json');
+
+class BotTypeOfCards extends ActivityHandler {
+    constructor() {
+        super();
+
+        this.onMembersAdded(async (context, next) => {
+            // Send welcome message when app installed
+            await this.sendWelcomeMessage(context);
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+
+        this.onMessage(async (context, next) => {
+            const text = context.activity.text;
+
+            // Create an array with the valid card options.
+            const suggestedCards = ['AdaptiveCard', 'HeroCard', 'ListCard', 'Office365', 'CollectionCard', 'SignIn', 'ThumbnailCard'];
+
+            // If the `text` is in the Array, a valid color was selected and send agreement.
+            if (suggestedCards.includes(text)) {
```

Comment:
Check if you have implemented all of these cards.

new Choice() { Value = "Adaptive Card", Synonyms = new List<string>() { "adaptive" } },
                    new Choice() { Value = "Hero Card", Synonyms = new List<string>() { "hero" } },
                    new Choice() { Value = "OAuth Card", Synonyms = new List<string>() { "oauth" } },
                    new Choice() { Value = "Signin Card", Synonyms = new List<string>() { "signin" } },
                    new Choice() { Value = "Thumbnail Card", Synonyms = new List<string>() { "thumbnail", "thumb" } },
                    new Choice() { Value = "List Cards", Synonyms = new List<string>() { "list" } },
                    new Choice() { Value = "Collections Cards", Synonyms = new List<string>() { "collections" } },
                    new Choice() { Value = "Connector Cards", Synonyms = new List<string>() { "office" } },

Reply:
I will update this comment before 4pm. Once after test is done.